lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -211,16 +213,22 @@
    * Closes all channels and releases all resources.
    */
   @Override
-  public void close() throws Exception {
+  public void close() {
 
     LOG.log(Level.FINE, "Closing netty transport socket address: {0}", this.localAddress);
 
     this.clientChannelGroup.close().awaitUninterruptibly();
     this.serverChannelGroup.close().awaitUninterruptibly();
-    this.acceptor.close().sync();
-    this.clientWorkerGroup.shutdownGracefully();
-    this.serverBossGroup.shutdownGracefully();
-    this.serverWorkerGroup.shutdownGracefully();
+
+    try {
+      this.acceptor.close().sync();
+    } catch (final Exception ex) {
+      LOG.log(Level.WARNING, "Error closing the acceptor channel for " + this.localAddress, ex);
```

Comment:
I am not sure what resources the acceptor channel holds here, other than the port it was listening on. Anyway, if we fail to close it, I am not sure we should stop the shutdown process.

In general, I think it is a good idea to make sure `.close()` never throws - similar to the C++ policy that destructors must never throw. The rationale here is that we are already in the process of cleaning up and on each stage we shall make the best effort and not prevent the subsequent stages from releasing the remaining resources.

That said, I think it is sensible to change the log message here to `SEVERE` level, but I really don't know what we can do better if we don't catch an exception here and let it fly outside of the method.

**P.S.** I am also not quite sure about the complete set of exceptions that could be raised here. The few exceptions that I saw in the logs were all related to attempts to close the port that had already been closed. So by catching the exceptions we're effectively making the `.close()` method idempotent.